uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/providers/IdentityProviderEndpointsAliasMockMvcTests.java

```diff
@@ -327,229 +373,351 @@ protected UpdateBase(final boolean aliasFeatureEnabled) {
+                    );
+                    assertThat(existingIdpWithoutAlias).isNotNull();
+                    assertThat(existingIdpWithoutAlias.getId()).isNotBlank();
+
+                    // perform update: set Alias ZID
+                    existingIdpWithoutAlias.setAliasZid(zone2.getId());
+                    final IdentityProvider<?> idpAfterUpdate = updateIdp(zone1, existingIdpWithoutAlias);
+                    assertThat(idpAfterUpdate.getAliasId()).isNotBlank();
+                    assertThat(idpAfterUpdate.getAliasZid()).isNotBlank();
+                    assertThat(zone2.getId()).isEqualTo(idpAfterUpdate.getAliasZid());
+
+                    // read alias IdP through alias id in original IdP
+                    final String id = idpAfterUpdate.getAliasId();
+                    final Optional<IdentityProvider<?>> idp = readIdpFromZoneIfExists(zone2.getId(), id);
+                    assertThat(idp).isPresent();
+                    final IdentityProvider<?> aliasIdp = idp.get();
+                    assertIdpReferencesOtherIdp(aliasIdp, idpAfterUpdate);
+                    assertOtherPropertiesAreEqual(idpAfterUpdate, aliasIdp);
+                }
 
-                // check if the returned IdP has a redacted relying party secret
-                assertRelyingPartySecretIsRedacted(updatedOriginalIdp);
-            }
+                @Test
+                void shouldReject_ReferencedZoneDoesNotExist() throws Exception {
+                    final IdentityZone zone = IdentityZone.getUaa();
+                    final IdentityProvider<?> existingIdp = createIdp(
+                            zone,
+                            buildUaaIdpWithAliasProperties(zone.getId(), null, null)
+                    );
 
-            @Test
-            void shouldAccept_ReferencedIdpNotExisting_ShouldCreateNewAliasIdp_UaaToCustomZone() throws Exception {
-                shouldAccept_ReferencedIdpNotExisting_ShouldCreateNewAliasIdp(IdentityZone.getUaa(), customZone);
-            }
+                    existingIdp.setAliasZid(UUID.randomUUID().toString()); // non-existing zone
 
-            @Test
-            void shouldAccept_ReferencedIdpNotExisting_ShouldCreateNewAliasIdp_CustomToUaaZone() throws Exception {
-                shouldAccept_ReferencedIdpNotExisting_ShouldCreateNewAliasIdp(customZone, IdentityZone.getUaa());
-            }
+                    shouldRejectUpdate(zone, existingIdp, HttpStatus.UNPROCESSABLE_ENTITY);
+                }
 
-            private void shouldAccept_ReferencedIdpNotExisting_ShouldCreateNewAliasIdp(final IdentityZone zone1, final IdentityZone zone2) throws Exception {
-                final IdentityProvider<?> idp = createIdpWithAlias(zone1, zone2);
+                @Test
+                void shouldReject_AliasNotSupportedForIdpType_UaaToCustomZone() throws Exception {
+                    shouldReject_AliasNotSupportedForIdpType(IdentityZone.getUaa(), customZone);
+                }
```

Comment:
Looks good. Leaving comments to track my progress as this is challenging to read through.